hwtypes/smt_utils.py

```diff
@@ -0,0 +1,76 @@
+from . import SMTBit
+import operator
+from functools import partial, reduce
+import abc
+
+_or_reduce = partial(reduce, operator.or_)
+_and_reduce = partial(reduce, operator.and_)
+
+class FormulaConstructor:
+    @abc.abstractmethod
+    def serialize(self, ts, indent):
+        ...
+
+    @abc.abstractmethod
+    def to_hwtypes(self):
+        ...
+
+def _to_hwtypes(v):
+    if isinstance(v, FormulaConstructor):
+        return v.to_hwtypes()
+    return v
+
+def _value_to_str(v, ts, indent):
+    if isinstance(v, FormulaConstructor):
+        return v.serialize(ts, indent)
+    else:
+        return f"{ts}{v.value.serialize()}"
+
+def _op_to_str(vs, opname, ts, indent):
+    new_ts = ts + indent
+    return "\n".join([
+        f"{ts}{opname}(",
+        ",\n".join([_value_to_str(v, new_ts, indent) for v in vs]),
+        f"{ts})"
+    ])
+def _check(vs):
+    if not all(isinstance(v, FormulaConstructor) or isinstance(v, SMTBit) for v in vs):
```

Comment:
`isinstance(v, (FormulaConstructor, SMTBit))`
Performs:
`isinstance(v, FormulaConstructor) or isinstance(v, SMTBit))`

For reference `issubclass` also can take a tuple of types.

Reply:
Fixed